src/components/posts/PostList.js

```diff
@@ -1,32 +1,51 @@
-import React, { useContext, useEffect } from "react"
+import React, { useContext, useEffect, useState } from "react"
 import { PostContext } from "./PostProvider"
-import { Link, useParams } from "react-router-dom"
+import { useHistory, useParams } from "react-router-dom"
 import "./Post.css"
 
 export const PostList = () => {
-    const { posts, getPosts, getPostsByUserId } = useContext(PostContext)
+    const { posts, getPosts, getPostsByUserId, deletePost } = useContext(PostContext)
 
     const { user_id } = useParams()
+    const history = useHistory()
+
+    const [isLoading, setIsLoading] = useState(true)
 
     useEffect(() => {
         if (user_id > 0) {
             getPostsByUserId(user_id)
+                .then(() => setIsLoading(false))
         } else {
             getPosts()
```

Comment:
PostList.js is responsible for rendering the list of all posts, so there should be a 
```.then(() => setIsLoading(false))``` after getPosts() as well.